sklearn/linear_model/_bayes.py

```diff
@@ -205,6 +207,103 @@ def __init__(
         self.copy_X = copy_X
         self.verbose = verbose
 
+    def _check_params(self):
+        """Check validity of parameters and raise ValueError
+        or TypeError if not valid."""
+
+        check_scalar(
+            self.n_iter,
+            name="n_iter",
+            target_type=numbers.Integral,
+            min_val=1,
+        )
+
+        check_scalar(
+            self.tol,
+            name="tol",
+            target_type=numbers.Real,
+            min_val=0.0,
+            include_boundaries="neither",
+        )
+
+        check_scalar(
+            self.alpha_1,
+            name="alpha_1",
+            target_type=numbers.Real,
+            include_boundaries="neither",
+        )
+
+        check_scalar(
+            self.alpha_2,
+            name="alpha_2",
+            target_type=numbers.Real,
+            include_boundaries="neither",
+        )
+
+        check_scalar(
+            self.lambda_1,
+            name="lambda_1",
+            target_type=numbers.Real,
+            include_boundaries="neither",
+        )
+
+        check_scalar(
+            self.lambda_2,
+            name="lambda_2",
+            target_type=numbers.Real,
+            include_boundaries="neither",
+        )
+
+        check_scalar(
+            self.alpha_init,
+            name="alpha_init",
+            target_type=(numbers.Real, type(None)),
+            include_boundaries="neither",
+        )
```

Comment:
```suggestion
        if self.alpha_init is not None:
            check_scalar(
                self.alpha_init,
                name="alpha_init",
                target_type=numbers.Real,
                include_boundaries="neither",
            )
```

Reply:
We don't really intend to use `check_scalar` with `None`.
it would be easier to add an `if` statement.

Reply:
Makes sense. Updated.